### DISCUSSION TITLE
How to Reinitialize NiceSelect2 After a Livewire Component Update

### DISCUSSION THREAD

**Discussion by keH1** [livewire/livewire#7988] [1. Help] [answered]
Hello everyone, I've dug through the bottom of the documentation and tried everything I could find, but I can't seem to achieve the desired effect.

I have a specific form that operates on the principle where you select a value from a multi-select list, and then a block with 3 additional selects appears below it. Essentially, the number of values selected in the multi-select dictates how many blocks appear. For styling into a nice multi-select, I use the NiceSelect2 library. Upon accessing the page, JS code is executed:
```js
document.addEventListener("DOMContentLoaded", function (e) {
    // default
    var els = document.querySelectorAll(".selectize");
    els.forEach(function (select) {
        var options = {searchable: false, placeholder: 'Выберите значение', selectedtext: 'Выбрано'};
        new NiceSelect(select, options);
    });
});
```
and everything is beautifully rendered.
To have the component update and the blocks displayed on the multi-select, the following is written: 
```blade
<select wire:model.live.debounce.2000ms="binderIds"
                                    multiple="multiple"
                                    x-ref="binders"
                                    id="binders"
                                    class="selectize">
                                @foreach($bitumenBinders as $bitumenBinder)
                                    <option @if(in_array($bitumenBinder->id, $binderIds)) selected="true"
                                            @endif
                                            value="{{ $bitumenBinder->id }}">{{ $bitumenBinder->name
                                    }}</option>
                                @endforeach
                            </select>
``` 
thereby, on the backend, I assemble the blocks and then simply display them through a loop.
However, the problem is that after the component updates and a block is added, all selects disappear from the page, in other words, each of them needs to be applied again with: `new NiceSelect(select, options);`,
I tried subscribing to different hooks - half of the data on forms for Livewire2 in the third version, some hooks just don't exist.
I tried triggering a custom event through 
```php
public function updatedBinderIds($value): void
    {
        foreach ($this->binderIds as $binderId) {
            if (isset($this->binderBlocks[$binderId])) {
                continue;
            }
            $this->binderBlocks[$binderId][] = ['roadId' => '', 'supplierId' => '', 'price' => ''];
        }
        $this->dispatch('binder-updated');
    }
```
it works, but for some reason, when executing the same code in JS, the DOM doesn't change at all and somehow retains its old version rather than the new one with added elements, and this also doesn't work!
The only working method is to randomly generate a `wire:key` and initiate niceSelect2 through Alpine, which works 
```blade
<div wire:key="{{ rand() }}"
                             x-init="() => { initNiceSelect($refs.binders) }">
                            <label for="binders"
                                   class="text-gray-800 text-sm font-medium inline-block mb-2">Марка битумного
                                                                                               вяжущего</label>
                            <select wire:model.live.debounce.2000ms="binderIds"
                                    multiple="multiple"
                                    x-ref="binders"
                                    id="binders"
                                    class="selectize">
                                @foreach($bitumenBinders as $bitumenBinder)
                                    <option @if(in_array($bitumenBinder->id, $binderIds)) selected="true"
                                            @endif
                                            value="{{ $bitumenBinder->id }}">{{ $bitumenBinder->name
                                    }}</option>
                                @endforeach
                            </select>
                        </div>
```
here is x-init function
```js
document.addEventListener('alpine:init', () => {
    Alpine.data('selects', () => ({
        initNiceSelect(element) {
            var options = {searchable: true, searchtext: 'Найти', placeholder: 'Выберите значение', selectedtext: 'Выбрано'};
            new NiceSelect(element, options);
        },
        showError(element) {
            console.log(element);
            new NiceSelect(element);
        },
    }));
});
```
but in the context of future tasks, I will need the key to be non-random, which means this path is closed for me.

I've only started learning Livewire recently, perhaps there are some methods, but I've tried all the options I found in the community answers and it feels like hardly anyone has implemented something similar, I ask for your help, dear community, perhaps someone will lead me to a clever thought.

**Comment by danie-ramdhani** [comment, marked as answer]
have you tried use `wire:ignore` in you NiceSelect2 root element? (https://livewire.laravel.com/docs/wire-ignore)

**Reply by keH1** [reply]
Wow, how did I not notice wire:ignore before? I tried all sorts of element updates, and the solution turned out to be incredibly simple. Thank you so much, everything now works as needed.
